requests_mv_integrations/request_mv_integration.py

```diff
@@ -0,0 +1,1774 @@
+        retry_tries=-1,  # default: -1 (indefinite)
+        retry_delay=0,
+        retry_max_delay=None,
+        retry_backoff=0,
+        retry_jitter=0,
+        request_label=None
+    ):
+        """Request Retry
+
+        Args:
+            call_func: the function to execute.
+            request_retry_excps: A tuple of exceptions to catch.
+            fargs: the positional arguments of the function to execute.
+            fkwargs: the named arguments of the function to execute.
+            timeout: (optional) How long to wait for the server to send
+                data before giving up.
+            request_retry_func: (optional) Retry alternative to request_retry_excps.
+
+            retry_tries: the maximum number of attempts.
+                default: -1 (infinite).
+            retry_delay: initial delay between attempts.
+                default: 0.
+            retry_max_delay:the maximum value of delay.
+                default: None (no limit).
+            retry_backoff: multiplier applied to delay between attempts.
+                default: 1 (no backoff).
+            retry_jitter: extra seconds added to delay between attempts.
+                default: 0.
+            request_label: Label
+
+        Returns:
+
+        """
+        request_retry_extra = {
+            'timeout': timeout
+        }
+
+        request_retry_extra.update({
+            'request_retry_http_status_codes': self.request_retry_http_status_codes
+        })
+
+        if self.request_retry_excps is not None:
+            request_retry_excp_names = [excp.__name__ for excp in list(self.request_retry_excps)]
+            request_retry_extra.update({
+                'request_retry_excps': request_retry_excp_names
+            })
+
+        if request_retry_func is not None:
+            request_retry_func_name = request_retry_func.__name__
+            request_retry_extra.update({
+                'request_retry_func': request_retry_func_name
+            })
+
+        if request_retry_excps_func is not None:
+            request_retry_excps_func_name = request_retry_excps_func.__name__
+            request_retry_extra.update({
+                'request_retry_excps_func': request_retry_excps_func_name
+            })
+
+        self.logger.debug(
+            "Request Retry: Start: {}".format(
+                request_label if request_label else ""
+            ),
+            extra=request_retry_extra
+        )
+
+        args = fargs if fargs else list()
+        kwargs = fkwargs if fkwargs else dict()
+
+        request_url = kwargs['request_url'] if kwargs and 'request_url' in kwargs else ""
+
+        _attempts = 0
+
+        self.retry_tries = retry_tries
+        self.retry_backoff = retry_backoff
+
+        _tries, _delay, _timeout = retry_tries, retry_delay, timeout
+        while _tries:
+            _attempts += 1
+
+            fkwargs['timeout'] = _timeout
+            request_func = partial(call_func, *args, **kwargs)
```

Comment:
This usage of partial should really be cleaned up. Its hard to follow and seeming unneeded. If refactoring the need away is not addressable here I suggest creating the partial in the calling function and only passing the request_func down. The kwarg should be overrideable down here in the function call below